internal/environment/checks.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package environment
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/corazawaf/coraza/v3/debuglog"
+)
+
+// IsDirWritable is a helper function to check if the WAF has access to the filesystem
+func IsDirWritable(logger debuglog.Logger, dir string) error {
+	if !HasAccessToFS {
```

Comment:
I'd rather decouple this from the other as we are mixing two concerns here. Also it is possible that for `tinygo.wasm` we just return false.

Reply:
I split the function based on the build tag and wrapped it checking HasAccessToFS. PTAL